ega-data-api-dataedge/src/main/java/eu/elixir/ega/ebi/dataedge/rest/FileController.java

```diff
@@ -15,306 +15,167 @@
 public class FileController {
 
     @Autowired
-    private FileService fileService;
+    private AuthenticationService authenticationService;
 
     @Autowired
-    private AuthenticationService authenticationService;
+    private PermissionsService permissionsService;
+
+    @Autowired
+    private NuFileService nuFileService;
 
     /**
      * Writes a requested file, or part of file from the FileService to the supplied
      * response stream.
-     *
-
-     */
-
-    /**
-     *
-     * @param fileId            ELIXIR id of the requested file.
-     * @param destinationFormat Requested destination format, either 'plain', 'aes',
-     *                          or file extension.
-     * @param destinationKey    Encryption key that the result file will be
-     *                          encrypted with.
-     * @param destinationIV     Initialization Vector for for destination file, used
-     *                          when requesting a partial AES encrypted file.
-     * @param startCoordinate   Start coordinate when requesting a partial file.
-     * @param endCoordinate     End coordinate when requesting a partial file.
-     * @param range
-     * @param request           Unused.
-     * @param response          Response stream for the returned data.
      */
     @RequestMapping(value = "/{fileId}", method = GET)
-    public void getFile(@PathVariable String fileId,
-                        @RequestParam(value = "destinationFormat", required = false, defaultValue = "plain") String destinationFormat,
-                        @RequestParam(value = "destinationKey", required = false, defaultValue = "") String destinationKey,
-                        @RequestParam(value = "destinationIV", required = false) String destinationIV,
-                        @RequestParam(value = "startCoordinate", required = false, defaultValue = "0") long startCoordinate,
-                        @RequestParam(value = "endCoordinate", required = false, defaultValue = "0") long endCoordinate,
-                        @RequestHeader(value = "Range", required = false, defaultValue = "") String range,
-                        HttpServletRequest request,
-                        HttpServletResponse response) {
-		String sessionId= Strings.isNullOrEmpty(request.getHeader("Session-Id"))? "" : request.getHeader("Session-Id") + " ";
-		log.info(sessionId + "Get file request started");
-		
-        if (range.length() > 0 && range.startsWith("bytes=") && startCoordinate == 0 && endCoordinate == 0) {
-            String[] ranges = range.substring("bytes=".length()).split("-");
-            startCoordinate = Long.valueOf(ranges[0]);
-            endCoordinate = Long.valueOf(ranges[1]) + 1; // translate into exclusive end coordinate
-        }
+    public ResponseEntity<StreamingResponseBody> getFile(@PathVariable String fileId,
+                                                         @RequestHeader(value = "Range", required = false, defaultValue = "") String range,
```

Comment:
Can you check if it can be converted to HttpRange at this point? Or later. as the HttpRange class already controls some of the edge cases parsing this field.

Reply:
I can use it but it changes the behaviour, if the user has an invalid range then HttpRange limits it to the size of the file so it always succeeds. Do you want that behaviour?

Reply:
Yes, if the range can't be parse properly, then as per RFC the most normal behaviour is to serve the file (in other words ignore the header) besides we will still check that the range is inside the contents and if the user is asking bytes outside of the range of the file we will respond as a non satisfiable range.

Reply:
>we will still check that the range is inside the contents and if the user is asking bytes outside of the range of the file we will respond as a non satisfiable range.

If I use HttpRange then it's not possible, it parses the header but will not tell me what the real values are that it parsed, it always restricts them to what the maximum length of the file is. Look at the implementation: https://github.com/spring-projects/spring-framework/blob/06e352822abc1eb0f5858ca9e79e38f8958e3c63/spring-web/src/main/java/org/springframework/http/HttpRange.java#L261

Reply:
Ok then, let's leave it at this, though we may need to revisit this in the future to make the parse more robust.